var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -242,10 +242,13 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     patch('piclibs.patch', when='+piclibs')
     patch('gcc-backport.patch', when='@4.7:4.9.2,5:5.3')
 
-    # Backport libsanitizer patch for glibc >= 2.31 and 8.1.0 <= gcc <= 9.2.0
+    # Backport libsanitizer patch for glibc >= 2.31 and 5.3.0 <= gcc <= 9.2.0
     # https://bugs.gentoo.org/708346
-    patch('glibc-2.31-libsanitizer-1.patch', when='@8.1.0:8.3.99,9.0.0:9.2.0')
+    patch('glibc-2.31-libsanitizer-1.patch', when='@7.1.0:7.4.99,8.1.0:8.3.99,9.0.0:9.2.0')
+    patch('glibc-2.31-libsanitizer-1-gcc-6.patch', when='@5.3.0:5.5.99,6.1.0:6.5.99')
     patch('glibc-2.31-libsanitizer-2.patch', when='@8.1.0:8.3.99,9.0.0:9.2.0')
+    patch('glibc-2.31-libsanitizer-2-gcc-6.patch', when='@5.3.0:5.5.99,6.1.0:6.5.99')
+    patch('glibc-2.31-libsanitizer-2-gcc-7.patch', when='@7.1.0:7.4.99')
```

Comment:
```suggestion
    patch('glibc-2.31-libsanitizer-1.patch', when='@7.1.0:7.4.0,8.1.0:8.3.0,9.0.0:9.2.0')
    patch('glibc-2.31-libsanitizer-1-gcc-6.patch', when='@5.3.0:5.5.0,6.1.0:6.5.0')
    patch('glibc-2.31-libsanitizer-2.patch', when='@8.1.0:8.3.0,9.0.0:9.2.0')
    patch('glibc-2.31-libsanitizer-2-gcc-6.patch', when='@5.3.0:5.5.0,6.1.0:6.5.0')
    patch('glibc-2.31-libsanitizer-2-gcc-7.patch', when='@7.1.0:7.4.0')
```
Since there are no patch releases, I would suggest to not use the usual 99 workaround and just go for the actual versions (see above). I don't have a build machine with glibc 2.31 available, so I cannot reproduce the issue. LGTM, though.

Reply:
I've changed the version to not use the 99 workaround, including the use that was found in the code before the patch. If that was not intended, let me know and I'll change it back.